### PR TITLE
(FACT-2504) Fix RSpec/MultipleExpectations offences

### PR DESCRIPTION
**Pull request by oanatmaria**

### CODE REVIEW

**Review by Filipovici-Andrei — CHANGES_REQUESTED** [review]

File: spec/custom_facts/core/resolvable_spec.rb [hunk 1]
```diff
@@ -49,15 +48,15 @@ def initialize(name)
   describe 'timing out' do
     it 'uses #limit instead of #timeout to determine the timeout period' do
       expect(resolvable).not_to receive(:timeout)
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply spec/custom_facts/core/resolvable_spec.rb]: RSpec/MessageSpies is another rubocop offence that will be handle under another ticket

File: spec/custom_facts/util/collection_spec.rb [hunk 2]
```diff
@@ -41,30 +41,29 @@
...
 
         expect(fact).to receive(:add)
 
         collection.add(:myname) {}
       end
 
       it 'discards resolutions that throw an exception when added' do
-        expect(LegacyFacter).to receive(:warn).with(/Unable to add resolve .* kaboom!/)
+        allow(LegacyFacter).to receive(:warn).with(/Unable to add resolve .* kaboom!/)
         expect do
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply spec/custom_facts/util/collection_spec.rb]: RSpec/MessageSpies is another rubocop offence that will be handle under another ticket

File: spec/custom_facts/util/loader_spec.rb [hunk 3]
```diff
@@ -189,15 +184,16 @@ def loader_from(places)
         allow(Kernel).to receive(:load).with(f)
       end
 
+      expect(Kernel).to receive(:load).with('/one/dir/testing.rb')
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply spec/custom_facts/util/loader_spec.rb]: RSpec/MessageSpies is another rubocop offence that will be handle under another ticket

File: spec/custom_facts/util/loader_spec.rb [hunk 4]
```diff
@@ -189,15 +184,16 @@ def loader_from(places)
...
     it "does not load files that don't end in '.rb'" do
       loader = LegacyFacter::Util::Loader.new
-      expect(loader).to receive(:search_path).and_return %w[/one/dir]
+      allow(loader).to receive(:search_path).and_return %w[/one/dir]
       allow(FileTest).to receive(:file?).and_return false
-      expect(FileTest).to receive(:file?).with('/one/dir/testing.rb').and_return false
-      expect(File).not_to receive(:readable?).with('/one/dir/testing')
+      allow(FileTest).to receive(:file?).with('/one/dir/testing.rb').and_return false
+
       expect(Kernel).not_to receive(:load)
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply spec/custom_facts/util/loader_spec.rb]: RSpec/MessageSpies is another rubocop offence that will be handle under another ticket

File: spec/custom_facts/util/loader_spec.rb [hunk 5]
```diff
@@ -220,49 +216,57 @@ def loader_from(places)
       allow(Dir).to receive(:glob).with('/two/dir/*.rb').and_return %w[/two/dir/c.rb /two/dir/d.rb]
 
       %w[/one/dir/a.rb /one/dir/b.rb /two/dir/c.rb /two/dir/d.rb].each do |f|
-        expect(FileTest).to receive(:file?).with(f).and_return true
+        allow(FileTest).to receive(:file?).with(f).and_return true
         expect(Kernel).to receive(:load).with(f)
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply spec/custom_facts/util/loader_spec.rb]: RSpec/MessageSpies is another rubocop offence that will be handle under another ticket

File: spec/custom_facts/util/loader_spec.rb [hunk 6]
```diff
@@ -220,49 +216,57 @@ def loader_from(places)
...
+      allow(Kernel).to receive(:load).with('/one/dir/b.rb')
 
       # c.rb is a directory
-      expect(Dir).to receive(:glob).with('/two/dir/*.rb').and_return %w[/two/dir/c.rb /two/dir/d.rb]
-      expect(FileTest).to receive(:file?).with('/two/dir/c.rb').and_return false
-      expect(FileTest).to receive(:file?).with('/two/dir/d.rb').and_return true
+      allow(Dir).to receive(:glob).with('/two/dir/*.rb').and_return %w[/two/dir/c.rb /two/dir/d.rb]
+      allow(FileTest).to receive(:file?).with('/two/dir/c.rb').and_return false
+      allow(FileTest).to receive(:file?).with('/two/dir/d.rb').and_return true
       expect(Kernel).to receive(:load).with('/two/dir/d.rb')
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply spec/custom_facts/util/loader_spec.rb]: RSpec/MessageSpies is another rubocop offence that will be handle under another ticket

File: spec/facter/facter_spec.rb [hunk 7]
```diff
@@ -327,7 +325,7 @@
 
       it "doesn't log anything" do
         expect_any_instance_of(Facter::Log).not_to receive(:debug).with(message)
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply spec/facter/facter_spec.rb]: RSpec/MessageSpies is another rubocop offence that will be handle under another ticket

File: spec/custom_facts/util/fact_spec.rb [hunk 8]
```diff
@@ -192,20 +192,23 @@ def suitable?
...
   describe '#flush' do
-    subject do
-      Facter::Util::Fact.new(:foo)
-    end
+    subject(:fact) { Facter::Util::Fact.new(:foo) }
 
-    it 'invokes #flush on all resolutions' do
+    it 'invokes #flush on simple resolutions' do
       simple = fact.add(type: :simple)
       expect(simple).to receive(:flush)
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply spec/custom_facts/util/fact_spec.rb]: RSpec/MessageSpies is another rubocop offence that will be handle under another ticket

File: spec/custom_facts/util/loader_spec.rb [hunk 9]
```diff
@@ -165,20 +162,18 @@ def loader_from(places)
     it 'loads any files in the search path with names matching the fact name' do
       loader = loader_from(search_path: %w[/one/dir /two/dir])
 
-      expect(loader).to receive(:search_path).and_return %w[/one/dir /two/dir]
+      allow(loader).to receive(:search_path).and_return %w[/one/dir /two/dir]
       allow(FileTest).to receive(:file?).and_return false
       allow(FileTest).to receive(:file?).with('/one/dir/testing.rb').and_return true
 
       expect(Kernel).to receive(:load).with('/one/dir/testing.rb')
```
[the remark this answers is not in the document]
Comment (oanatmaria) [reply spec/custom_facts/util/loader_spec.rb]: RSpec/MessageSpies is another rubocop offence that will be handle under another ticket

File: spec/facter/query_parser_spec.rb [hunk 10]
```diff
@@ -14,8 +14,8 @@
 
       matched_facts = Facter::QueryParser.parse(query_list, loaded_facts)
 
-      expect(matched_facts.size).to eq(1)
-      expect(matched_facts.first.fact_class).to eq(os_name_class)
+      expect(matched_facts).to be_an_instance_of(Array).and \
```
[the remark this answers is not in the document]
Comment (IrimieBogdan) [reply spec/facter/query_parser_spec.rb]: For complex objects I would go with the version in the PR, it is easier to read.

File: spec/custom_facts/core/aggregate_spec.rb [hunk 11]
```diff
@@ -80,15 +80,21 @@
       end
 
       output = aggregate_res.value
-      expect(output).to include 'foo'
-      expect(output).to include 'foo bar'
+      expect(output).to be_an_instance_of(Array).and contain_exactly('foo', 'foo bar')
```
[the remark this answers is not in the document]
Comment (IrimieBogdan) [reply spec/custom_facts/core/aggregate_spec.rb]: For strings, the shorter version is succinct and I would go with the approach @Filipovici-Andrei  proposed

**Review by Filipovici-Andrei — APPROVED** [review]

**Pull request merged**